__tests__/posts.test.js

```diff
@@ -0,0 +1,101 @@
+const setup = require('../data/setup');
+const request = require('supertest');
+const app = require('../lib/app');
+const GithubUser = require('../lib/models/GithubUser');
+const pool = require('../lib/utils/pool');
+
+const agent = request.agent(app);
+jest.mock('../lib/utils/github.js');
+
+describe('github posts route', () => {
+  beforeEach(() => {
+    return setup(pool);
+  });
+
+  afterAll(() => {
+    pool.end();
+  });
+
+  it('creates a post', async () => {
+    await GithubUser.insert({
+      username: 'fake_github_user',
+      email: 'not-real@example.com',
+    });
+
+    await agent.get('/api/v1/github/login/callback?code=42').redirects(1);
+
+    const res = await agent
+      .post('/api/v1/posts')
+      .send({ text: 'This is a post' });
+
+    expect(res.body).toEqual({
+      id: 1,
+      text: 'This is a post',
+      username: 'fake_github_user',
+    });
+  });
+
+  it('tests that post length is limited to 255 chars', async () => {
+    await GithubUser.insert({
```

Comment:
We shouldn't need to use GithubUser.insert. You should instead use: 
login/callback route which will check if a user exists or not.